ppa/archive/management/commands/split_marc.py

```diff
@@ -0,0 +1,58 @@
+from collections import Counter
+from io import StringIO
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+import pymarc
+
+from ppa.archive.gale import get_marc_storage
+
+
+class Command(BaseCommand):
+    """Split MARC records out so they are easily accessible by item id
+    for import and Zotero metadata."""
+
+    help = __doc__
+
+    #: normal verbosity level
+    v_normal = 1
+    verbosity = v_normal
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "marcfiles", nargs="+", help="List of MARC files to read and split"
+        )
+
+    def handle(self, *args, **kwargs):
+        # initialize pairtree storage for storing the split out
+        # marc files
+        marc_store = get_marc_storage()
+        stats = Counter()
+        for filepath in kwargs["marcfiles"]:
+            stats["files"] += 1
+            with open(filepath, "rb") as marcfile:
+                reader = pymarc.MARCReader(marcfile, to_unicode=True)
+                for record in reader:
+                    # Gale item id used for API is only available in the
+                    # access url; split the url and get the id
+                    gale_id = record["856"]["u"].split("/")[5]
+                    # create an object in the pairtree
+                    pmarc = marc_store.get_object(gale_id, create_if_doesnt_exist=True)
+                    # use pymarc to export as JSON (more compact & faster than XML)
+                    output = StringIO()
+                    writer = pymarc.JSONWriter(output)
+                    writer.write(record)
+                    writer.close(close_fh=False)  # important for valid json!
```

Comment:
is it `writer.close()` that's important, or passing `close_fh=False`? i'm kinda unclear how this works – if the file handle isn't closed, would it automatically be closed once this iteration of the forloop completes and `writer` goes out of scope?

Reply:
Ah, I see how my comment is unclear.

From the pymarc documentation/examples, closing the writer is important to make sure the json is valid. In this case, we need to not close the file handle because that's the output buffer we're writing to.